x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/MutableSearchResponse.java

```diff
@@ -103,18 +113,35 @@ synchronized void updatePartialResponse(
         this.totalHits = totalHits;
         this.reducedAggsSource = reducedAggs;
         this.reducePhase = reducePhase;
+        if (isFinalLocalReduce) {
+            if (localClusterSearchCompleted == false
```

Comment:
is this flag still needed? I think you could have an assert that verifies that final reduction is performed once and remove the flag. Am I missing something?

Reply:
Yeah, good question. I did try to remove it in my last PR rewrite, but I thought I had found it was still needed. But at your prompt, I removed it again just now and tested it again and now I don't find any case where it's needed, so I'll remove it in the next commit.